rule/error_strings.go

```diff
@@ -178,21 +178,25 @@ func (lintErrorStrings) checkArg(expr *ast.CallExpr, arg int) (s *ast.BasicLit,
 func lintErrorString(s string) (isClean bool, conf float64) {
 	const basicConfidence = 0.8
 	const capConfidence = basicConfidence - 0.2
+
 	first, firstN := utf8.DecodeRuneInString(s)
 	last, _ := utf8.DecodeLastRuneInString(s)
 	if last == '.' || last == ':' || last == '!' || last == '\n' {
 		return false, basicConfidence
 	}
+
```

Comment:
Variables about` first` could be moved after the `if` about `last` one

The performance difference will be imperceptible, but at least first and firstname will be defined just before they are used


```suggestion

 	last, _ := utf8.DecodeLastRuneInString(s)
 	if last == '.' || last == ':' || last == '!' || last == '\n' {
 		return false, basicConfidence
 	}

 	first, firstN := utf8.DecodeRuneInString(s)
```